### PR TITLE
Channel share

### PR DESCRIPTION
**Pull request by dzucconi**
share

### CODE REVIEW

File: apps/channel/components/share/view.coffee [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  initialize: ({ channel, text }) ->
+    @channel = channel
+    @state = new Backbone.Model
+      status: 'inactive'
+      url: encodeURIComponent(APP_URL + CURRENT_PATH)
+      text: encodeURIComponent(text)
+
+    @listenTo @state, 'change', @render
+
+  toggle: (e) ->
```
Comment (broskoski) [inline apps/channel/components/share/view.coffee]: 👌

File: apps/channel/components/share/view.coffee [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        .generateShareLink()
+        .then () =>
+          @state.set
+            status: 'active'
+            url: @channel.shareHref()
+
+        .catch () =>
+          @state.set 'status', 'error'
+
+    else if @channel.get('status') is 'private'
```
Comment (broskoski) [inline apps/channel/components/share/view.coffee]: Am I following this right? If a channel already has a share link, it just shows the menu

**Comment by broskoski** [comment]
This is really nice, well done 🎨

File: apps/channel/components/share/view.coffee (hunk 2 above)
Comment (dzucconi) [reply apps/channel/components/share/view.coffee]: Yeah it just skips generation.

**Pull request merged**